### DISCUSSION TITLE
el-table无法监听内部滚动条

### DISCUSSION THREAD

**Discussion by darksunnydong** [element-plus/element-plus#15943] [Feature Request]
el-table的源码中对滚动条开放了4个api
    scrollBarRef: import("vue").Ref<any>;
    scrollTo: (options: number | ScrollToOptions, yCoord?: number | undefined) => void;
    setScrollLeft: (left?: number | undefined) => void;
    setScrollTop: (top?: number | undefined) => void;
其中scrollBarRef是滚动条的ref，源码中滚动条的ref是固定的，并不能动态的传入。如果一个页面上只有一个表格的话，这样监听表格的滚动条是没问题的，但实际应用中，往往一个页面上有多个el-table，这个时候表格的ref就不唯一了，dom操作监听表格的滚动条就无法实现。

希望官方能优化一下el-table的滚动条监听事件。

或者有大佬能指点一下如何解决这个问题，有偿

**Comment by xiaowangxiaowang256256** [comment]
如何解决一个页面，多个表格的滚动条的ref？
1、给每个表格添加一个独立的ref
2、根据业务 去控制你要使用的表格的 ref（由于每个表格都有独立的ref，则dom操作监听 表格的滚动条 就可实现）
![code-1](https://github.com/element-plus/element-plus/assets/111274524/6d9d83b2-4df2-480e-85d9-ceb5ee5a10b6)
![code](https://github.com/element-plus/element-plus/assets/111274524/c53125bf-58b4-4b87-acad-632d9fd9727c)

==============================
这样使用，我个人觉得 更加自由一些。希望我的回复能够对你有所帮助

**Reply by darksunnydong** [reply]
感谢您的回复，你提供的实例中，是对表格横向滚动进行设置，调用了el-table提供的api。我实际遇到的问题是，需要监听表格当前滚动条的位置（横向，纵向），并且要能动态设置表格的滚动条位置（横向，纵向）。
为此，可能需要新增一个事件函数和一个表格方法，表格scroll-change()和setScroll()

因为目前没有相应的事件和方法，所以只能对el-table的内部滚动条dom进行操作，使用起来比较麻烦

**Reply by xiaowangxiaowang256256** [reply]
**你的需求：**
1、需要监听表格当前滚动条的位置（横向，纵向）
2、并且要能动态设置表格的滚动条位置（横向，纵向）

**我的解决方案：**
1、tableRef.value?.scrollBarRef.wrapRef.scrollLeft //返回当前视图中的实际元素的左边缘和左边缘之间的距离，获取当前滚动条的位置（横向）
tableRef.value?.scrollBarRef.wrapRef.scrollTop  //返回当前视图中的实际元素的顶部边缘和顶部边缘之间的距离，获取当前滚动条的位置（纵向）
![77](https://github.com/element-plus/element-plus/assets/111274524/47a28c0a-4212-499d-93c9-c34df3a38237)

2、tableRef.value?.scrollTo(100, 200) //设置表格的滚动条位置（横向，纵向）
![77](https://github.com/element-plus/element-plus/assets/111274524/d2abfb15-25bd-4c14-9436-b7467f7112dc)

3、代码截图
![image](https://github.com/element-plus/element-plus/assets/111274524/4e565655-66cd-4591-9e2a-eee1da256b39)

这个方法是，对el-table的内部滚动条dom进行操作。是否能解决你的问题。

**Comment by elarbor** [comment]
表格就该用一个组件，就叫`<el-table>` 。现在是两个[`<el-table>` ](https://element-plus.org/zh-CN/component/table.html)和[`<el-table-v2>`](https://element-plus.org/zh-CN/component/table-v2.html) 。

用type区分更好一点：
`<el-table type="default">` 就是原来的`<el-table>`组件
`<el-table type="virtualized">`  就是原来的`<el-table-v2>` 组件
`<el-table type="第三种">`  就是以后的`<第三种>` 组件。    ......
